src/utils/mesh_utils.ts

```diff
@@ -9,30 +9,71 @@ import {
     ValidationResults,
     WSClient,
 } from '@0x/mesh-rpc-client';
+import axios from 'axios';
 import * as _ from 'lodash';
 
-import { ZERO } from '../constants';
+import {
+    MESH_HTTP_DEFAULT_ORDER_BATCH_SIZE,
+    MESH_WEBSOCKET_DEFAULT_ORDER_BATCH_SIZE,
+    MESH_WEBSOCKET_ORDER_BATCH_LIMIT,
+    ZERO,
+} from '../constants';
 import { ValidationErrorCodes } from '../errors';
 import { logger } from '../logger';
 import { AddedRemovedUpdate, APIOrderWithMetaData } from '../types';
 
+import { utils } from './utils';
+
 export const meshUtils = {
     addOrdersToMeshAsync: async (
+        meshHttpEndpoint: string,
         meshClient: WSClient,
         orders: SignedOrder[],
-        batchSize: number = 100,
     ): Promise<ValidationResults> => {
-        // Mesh rpc client can't handle a large amount of orders. This results in a fragmented
-        // send which Mesh cannot accept.
+        // Mesh rpc client can't handle a large amount of orders over websocket. This results in a fragmented
+        // send which Mesh cannot accept. We check how many orders are being added, and use HTTP if there are
+        // too many.
+        const shouldSendViaHttp = orders.length > MESH_WEBSOCKET_ORDER_BATCH_LIMIT;
+        const batchSize = shouldSendViaHttp
+            ? MESH_HTTP_DEFAULT_ORDER_BATCH_SIZE
+            : MESH_WEBSOCKET_DEFAULT_ORDER_BATCH_SIZE;
         const validationResults: ValidationResults = { accepted: [], rejected: [] };
         const chunks = _.chunk(orders, batchSize);
         for (const chunk of chunks) {
-            const results = await meshClient.addOrdersAsync(chunk as any);
+            const results = shouldSendViaHttp
```

Comment:
This is another artifact of not abstracting away the transport to use within a simple Mesh client interface.